# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1220892353 14400
#      Mon Sep 08 12:45:53 2008 -0400
# Node ID 7d19f4f70c613ca006e7062a4dba82b5a7c20a8d
# Parent  83646120b54c9c69b831236c65e32e2fb37c03fd
ls-oct-ascii.{h,cc} (extract_keyword): accept \r as line ending character

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,17 @@
+2008-09-08  John W. Eaton  <jwe@octave.org>
+
+	* ls-oct-ascii.cc (std::string extract_keyword (std::istream&,
+	const char *, const bool)): Also accept \r as line ending character.
+	* ls-oct-ascii.h (template <class T> bool extract_keyword
+	(std::istream&, const char *, T&, const bool)): Likewise.
+	(template <class T> bool extract_keyword (std::istream&, const
+	string_vector&, std::string&, T&, const bool)): Likewise.
+
 2008-09-03  Ben Abbott <bpabott@mac.com>
 
 	* mappers.cc: Use eps(single) as tolerance for single precesion test.
 
 2008-09-03  John W. Eaton  <jwe@octave.org>
 
 	* variables.cc (mlock): Lock current function on the call stack.
 	(Fmlock): Lock caller.
diff --git a/src/ls-oct-ascii.cc b/src/ls-oct-ascii.cc
--- a/src/ls-oct-ascii.cc
+++ b/src/ls-oct-ascii.cc
@@ -105,31 +105,31 @@ extract_keyword (std::istream& is, const
 	  bool match = (tmp.compare (0, strlen (keyword), keyword) == 0);
 
 	  if (match)
 	    {
 	      std::ostringstream value;
 	      while (is.get (c) && (c == ' ' || c == '\t' || c == ':'))
 		; // Skip whitespace and the colon.
 
-	      if (c != '\n')
+	      if (c != '\n' && c != '\r')
 		{
 		  value << c;
-		  while (is.get (c) && c != '\n')
+		  while (is.get (c) && c != '\n' && c != '\r')
 		    value << c;
 		}
 
 	      retval = value.str ();
 	      break;
 	    }
 	  else if (next_only)
 	    break;
 	  else
 	    {
-	      while (is.get (c) && c != '\n')
+	      while (is.get (c) && c != '\n' && c != '\r')
 		; // Skip to end of line.
 	    }
 	}
     }
 
   int len = retval.length ();
 
   if (len > 0)
diff --git a/src/ls-oct-ascii.h b/src/ls-oct-ascii.h
--- a/src/ls-oct-ascii.h
+++ b/src/ls-oct-ascii.h
@@ -94,21 +94,21 @@ extract_keyword (std::istream& is, const
 	  bool match = (tmp.compare (0, strlen (keyword), keyword) == 0);
 
 	  if (match)
 	    {
 	      while (is.get (c) && (c == ' ' || c == '\t' || c == ':'))
 		; // Skip whitespace and the colon.
 
 	      is.putback (c);
-	      if (c != '\n')
+	      if (c != '\n' && c != '\r')
 		is >> value;
 	      if (is)
 		status = true;
-	      while (is.get (c) && c != '\n')
+	      while (is.get (c) && c != '\n' && c != '\r')
 		; // Skip to beginning of next line;
 	      break;
 	    }
 	  else if (next_only)
 	    break;
 	}
     }
   return status;
@@ -156,21 +156,21 @@ extract_keyword (std::istream& is, const
 	      if (match)
 		{
 		  kw = keywords[i];
 
 		  while (is.get (c) && (c == ' ' || c == '\t' || c == ':'))
 		    ; // Skip whitespace and the colon.
 
 		  is.putback (c);
-		  if (c != '\n')
+		  if (c != '\n' && c != '\r')
 		    is >> value;
 		  if (is)
 		    status = true;
-		  while (is.get (c) && c != '\n')
+		  while (is.get (c) && c != '\n' && c != '\r')
 		    ; // Skip to beginning of next line;
 		  return status;
 		}
 	    }
 
 	  if (next_only)
 	    break;
 	}
